### ISSUE TITLE
Nan during training 

### ISSUE THREAD

**Issue by jgrietens** [yhenon/pytorch-retinanet#140]
Hi, 
I'm getting Nan's after a few batches of training. 

First the bb predictions go to zero and later they flip to Nans, as do the losses. 

Does anybody have any clue what might cause this ?

**Comment by Laudarisd** [comment]
I had the same issue. Decreasing learning rate might help.

**Comment by jgrietens** [comment]
The problem was fixed after reducing my learning rate and limiting the gradients.

**Comment by Laudarisd** [comment]
Great. Did you train it on  your own data-set?I trained on my own data but inference code doesn't work as it is mentioned in this open source.Would you mind to share your inference code for object detection on images or repository? It would be a great help for me.
